lib/monitoring/middleware/conjur_collector.rb

```diff
@@ -0,0 +1,65 @@
+require 'benchmark'
+require ::File.expand_path('../../pub_sub.rb', __FILE__)
+
+module Monitoring
+  module Middleware
+    class ConjurCollector
+      attr_reader :app, :registry, :pubsub
+
+      def initialize(app, options = {})
+        @app = app
+        @registry = options[:registry]
+        @pubsub = options[:pubsub]
+
+        init_test_metric
+      end
+
+      def call(env)
+        trace(env) { @app.call(env) }
+      end
+
+      protected
+
+      def init_test_metric
```

Comment:
I'm wondering if this should be consolidated under prometheus.rb? There is already a test metric there (not using pub/sub), plus I think it helps to think about it in a more generic sense, because presumably there will be metrics collected for events beyond just incoming requests